src/ws/session.c

```diff
@@ -500,7 +501,8 @@ pam_conv_func (int num_msg,
 static int
 open_session (pam_handle_t *pamh)
 {
-  struct passwd *buf = NULL;
+  /* This buffer can be static because it's used by pwd, which is also static */
+  static struct passwd *buf = NULL;
```

Comment:
To make things more legible, I would suggest moving this right next to the passwd decl above.